src/lookup-table/utils/dataTypeUtils.ts

```diff
@@ -0,0 +1,103 @@
+import { mapValues } from 'lodash'
+import { FieldMapType } from '../types'
+import { DateTime } from 'luxon'
+
+export const setDataTypes = (fieldMaps: FieldMapType[], rows: object[]) => {
+  // Determine type
+  const done: string[] = []
+  for (const row of rows) {
+    Object.entries(row).forEach(([key, value], index) => {
+      if (!done.includes(key)) {
+        const type = getType(value)
+        if (type !== null) {
+          fieldMaps[index].dataType = type
+          done.push(key)
+        }
+      }
+    })
+
+    if (done.length === fieldMaps.length) break
+  }
```

Comment:
It just checks the first value of each column to determine its type. However, we can have `null` values, so it iterates through all rows until it's found at least one non-null value for each field, then it breaks out of the loop.